Fix:

config.c:182:8: error: unknown type name 'locale_t'
config.c:186:6: error: implicit declaration of function 'newlocale' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
config.c:186:16: error: use of undeclared identifier 'LC_CTYPE_MASK'
config.c:193:15: error: implicit declaration of function 'toupper_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
config.c:205:12: error: implicit declaration of function 'strncasecmp_l' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

Reported to developers:

https://lists.gnu.org/archive/html/info-mtools/2020-11/msg00001.html

Can hopefully be removed in a future version.
--- sysincludes.h.orig	2018-12-09 17:28:55.000000000 -0600
+++ sysincludes.h	2020-11-09 03:06:14.000000000 -0600
@@ -347,6 +347,7 @@
 #ifdef HAVE_LOCALE_H
 # include <locale.h>
 #endif
+#include <xlocale.h>
 
 #ifdef USE_FLOPPYD
 
